libs/synchronization/include/hpx/synchronization/condition_variable.hpp

```diff
@@ -260,6 +240,109 @@ namespace hpx { namespace lcos { namespace local {
             return wait_until(lock, rel_time.from_now(), std::move(pred), ec);
         }
 
+        // 32.6.4.2, interruptible waits
+        template <typename Lock, typename Predicate>
+        bool wait(Lock& lock, stop_token stoken, Predicate pred,
+            error_code& ec = throws)
+        {
+            if (stoken.stop_requested())
+            {
+                return pred();
+            }
+
+            auto f = [&] {
+                std::unique_lock<mutex_type> l(mtx_.data_);
+                cond_.data_.notify_all(std::move(l), ec);
+            };
+            stop_callback<decltype(f)> cb(stoken, std::move(f));
+
+            while (!pred())
+            {
+                util::ignore_all_while_checking ignore_lock;
+                std::unique_lock<mutex_type> l(mtx_.data_);
+                if (stoken.stop_requested())
```

Comment:
That's a good question. I'm not sure - what do you think? Nico's jthread implementation locks it as well: https://github.com/josuttis/jthread/blob/master/source/condition_variable_any2.hpp#L227

Reply:
TBH, I have no idea with what the `stoken.stop_requested()` needs to be synchronized with in this case. Judging from empirical evidence (Line 248 and https://github.com/josuttis/jthread/blob/master/source/condition_variable_any2.hpp#L221), I think it need and must not be protected by the internal lock.

Reply:
Fair point. I'll move it out of the locked region.

Reply:
I think this check is performed to make sure the stop state is evaluated in case it was changed while the current thread was acquiring the lock. For this reason, I'll leave the check where it is.